integration-tests/deployment/keystone/ocr3config.go

```diff
@@ -0,0 +1,227 @@
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/chaintype"
+	"github.com/smartcontractkit/chainlink/v2/core/services/ocrcommon"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm"
+)
+
+type TopLevelConfigSource struct {
+	OracleConfig OracleConfigSource
+}
+
+type OracleConfigSource struct {
+	MaxQueryLengthBytes       uint32
+	MaxObservationLengthBytes uint32
+	MaxReportLengthBytes      uint32
+	MaxRequestBatchSize       uint32
+	UniqueReports             bool
+
+	DeltaProgressMillis               uint32
+	DeltaResendMillis                 uint32
+	DeltaInitialMillis                uint32
+	DeltaRoundMillis                  uint32
+	DeltaGraceMillis                  uint32
+	DeltaCertifiedCommitRequestMillis uint32
+	DeltaStageMillis                  uint32
+	MaxRoundsPerEpoch                 uint64
+	TransmissionSchedule              []int
+
+	MaxDurationQueryMillis       uint32
+	MaxDurationObservationMillis uint32
+	MaxDurationAcceptMillis      uint32
+	MaxDurationTransmitMillis    uint32
+
+	MaxFaultyOracles int
+}
+
+type NodeKeys struct {
+	EthAddress            string `json:"EthAddress"`
+	AptosAccount          string `json:"AptosAccount"`
+	AptosBundleID         string `json:"AptosBundleID"`
+	AptosOnchainPublicKey string `json:"AptosOnchainPublicKey"`
+	P2PPeerID             string `json:"P2PPeerID"`             // p2p_<key>
+	OCR2BundleID          string `json:"OCR2BundleID"`          // used only in job spec
+	OCR2OnchainPublicKey  string `json:"OCR2OnchainPublicKey"`  // ocr2on_evm_<key>
+	OCR2OffchainPublicKey string `json:"OCR2OffchainPublicKey"` // ocr2off_evm_<key>
+	OCR2ConfigPublicKey   string `json:"OCR2ConfigPublicKey"`   // ocr2cfg_evm_<key>
+	CSAPublicKey          string `json:"CSAPublicKey"`
+}
+
+type Orc2drOracleConfig struct {
+	Signers               [][]byte
+	Transmitters          []common.Address
+	F                     uint8
+	OnchainConfig         []byte
+	OffchainConfigVersion uint64
+	OffchainConfig        []byte
+}
+
+func (c Orc2drOracleConfig) MarshalJSON() ([]byte, error) {
+	alias := struct {
+		Signers               []string
+		Transmitters          []string
+		F                     uint8
+		OnchainConfig         string
+		OffchainConfigVersion uint64
+		OffchainConfig        string
+	}{
+		Signers:               make([]string, len(c.Signers)),
+		Transmitters:          make([]string, len(c.Transmitters)),
+		F:                     c.F,
+		OnchainConfig:         "0x" + hex.EncodeToString(c.OnchainConfig),
+		OffchainConfigVersion: c.OffchainConfigVersion,
+		OffchainConfig:        "0x" + hex.EncodeToString(c.OffchainConfig),
+	}
+
+	for i, signer := range c.Signers {
+		alias.Signers[i] = hex.EncodeToString(signer)
+	}
+
+	for i, transmitter := range c.Transmitters {
+		alias.Transmitters[i] = transmitter.Hex()
+	}
+
+	return json.Marshal(alias)
+}
+
+func GenerateOCR3Config(cfg OracleConfigSource, nca []NodeKeys) (Orc2drOracleConfig, error) {
```

Comment:
we can come back to it, but there should  be a way to can share OCR3/Capability registry management code in for example some deployment/capreg package.